pubsub/pubsub_test.go

```diff
@@ -143,10 +138,17 @@ func flatten(responses [][]string) []string {
 	return ret
 }
 
-func produceMessages(ctx context.Context, msgs []string, producer *Producer[testRequest, testResponse]) ([]*containers.Promise[testResponse], error) {
+func produceMessages(ctx context.Context, msgs []string, producer *Producer[testRequest, testResponse], useUniqueIdentifier, withInvalidEntries bool) ([]*containers.Promise[testResponse], error) {
```

Comment:
could you explain why useuniqueIdentifier is helpful?

Reply:
I had `useUniqueIdentifier` to show that this pubsub model should work when theres just one producer and `SetSelfHash`(or unique id) isn't used. 
But makes sense to remove it!